### PR TITLE
[SYCL][NFC] Increase test coverage for inline asm feature

### PR DESCRIPTION
**Pull request by fadeeval**
This PR adds inline_asm feature tests with branching and loop.

Signed-off-by: Aleksander Fadeev aleksander.fadeev@intel.com

### CODE REVIEW

File: sycl/test/inline-asm/asm_if.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// UNSUPPORTED: cuda
+// REQUIRES: gpu,linux
+// RUN: %clangxx -fsycl %s -DINLINE_ASM -o %t.out
+// RUN: %t.out
```
Comment (alexbatashev) [inline sycl/test/inline-asm/asm_if.cpp:4]: What happens if I run this on a machine without a GPU? Should this be like
```suggestion
// RUN: %GPU_RUN_PLACEHOLDER %t.out
```

File: sycl/test/inline-asm/asm_if.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+// RUN: %clangxx -fsycl %s -o %t.ref.out
+// RUN: %t.ref.out
+
+#include "include/asmhelper.h"
+#include <CL/sycl.hpp>
+
+using dataType = cl::sycl::cl_int;
```
Comment (alexbatashev) [inline sycl/test/inline-asm/asm_if.cpp:11]:
```suggestion
using DataType = cl::sycl::cl_int;
```

File: sycl/test/inline-asm/asm_if.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+template <typename T = dataType> struct KernelFunctor : WithOutputBuffer<T> {
+  KernelFunctor(size_t problem_size) : WithOutputBuffer<T>(problem_size) {}
```
Comment (alexbatashev) [inline sycl/test/inline-asm/asm_if.cpp:14]:
```suggestion
  KernelFunctor(size_t ProblemSize) : WithOutputBuffer<T>(ProblemSize) {}
```

File: sycl/test/inline-asm/asm_loop.cpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        {
+          C[wiID] = C[wiID] + B[wiID];
+        }
+#endif
+        });
+  }
+};
+
+int main() {
+  std::vector<dataType> inputA(DEFAULT_PROBLEM_SIZE),
```
Comment (alexbatashev) [inline sycl/test/inline-asm/asm_loop.cpp:61]:
```suggestion
  std::vector<DataType> InputA(DEFAULT_PROBLEM_SIZE),
```

File: sycl/test/inline-asm/asm_if.cpp [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+          else
+            output = 8;
+#endif
+          C[wiID] = output;
+        });
+  }
+};
+
+int main() {
+  KernelFunctor<> f(DEFAULT_PROBLEM_SIZE);
```
Comment (alexbatashev) [inline sycl/test/inline-asm/asm_if.cpp:43]:
```suggestion
  KernelFunctor<> Functor(DEFAULT_PROBLEM_SIZE);
```

**Review by alexbatashev — COMMENTED** [review]
Just a few stylistic nits to better match overall project code style. Please, apply my comments to other places in code when appropriate.

**Review by alexbatashev — APPROVED** [review]
LGTM

**Comment by AlexeySotkin** [comment]
Please squash the commits and fix the typo in the [commit message](https://github.com/intel/llvm/pull/2124#issue-449439143)

**Comment by alexbatashev** [comment]
> Please squash the commits and fix the typo in the [commit message](https://github.com/intel/llvm/pull/2124#issue-449439143)

I think, @bader can squash commits from GitHub GUI and use PR description as a commit message, which is kinda what he always does. There's no need to squash the commits manually, restart tests and waste CPU cycles.

**Pull request merged**